### DISCUSSION TITLE
ser2net service fails after reboot

### DISCUSSION THREAD

**Discussion by fabnavigator** [zwave-js/node-red-contrib-zwave-js#222] [Q&A] [answered]
I've had to use ser2net under version 7, and though it works, I've always had to manually restart the service after a reboot.  Maybe /dev/ttyUSB0 isn't ready when the service trys to start. I don't know. If you have any thoughts about how I can fix this, I would appreciate it.

**Comment by marcus-j-davies** [comment]
Hi @fabnavigator, Hope you're well.

If you can find the `ser2net.service` file (the thing that `systemd` uses to run it)
you can set a restart section I believe.

it may be here : `/usr/lib/systemd/system/ser2net.service`

```
[Service]
...
Restart=on-failure
RestartSec=5
```

This will restart the `ser2net` daemon after 5 seconds a crash on it occurs.

After making this change, you will need to update systemd
```
systemctl daemon-reload
systemctl enable ser2net
systemctl restart ser2net
```

**Comment by marcus-j-davies** [comment]
It may Evan be worth just running `systemctl enable ser2net` as that will instruct systemd to make sure it starts on boot.

it may not be enabled to auto start 😅

**Comment by fabnavigator** [comment]
Hi @marcus-j-davies,

I'm doing well. I hope you are too. I've been quiet becasue I've recently moved from an apartment to a house. I'm poised to finally reinstall all of my 30 or so z-wave devices. Lots of fun in the near future.

The service is enabled and the service starts on boot. I see this below after a reboot.
```
fabnav@docker-hass:~ $ sudo systemctl status ser2net
● ser2net.service - Serial port to network proxy
     Loaded: loaded (/lib/systemd/system/ser2net.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2022-06-07 17:45:34 EDT; 3min 6s ago
       Docs: man:ser2net(8)
   Main PID: 481 (ser2net)
      Tasks: 1 (limit: 8985)
        CPU: 78ms
     CGroup: /system.slice/ser2net.service
             └─481 /usr/sbin/ser2net -n -c /etc/ser2net.yaml -P /run/ser2net.pid

Jun 07 17:45:34 docker-hass systemd[1]: Starting Serial port to network proxy...
Jun 07 17:45:34 docker-hass systemd[1]: Started Serial port to network proxy.
Jun 07 17:45:34 docker-hass ser2net[481]: Invalid port name/number: Invalid data to parameter on line 7 column 0
Jun 07 17:45:34 docker-hass ser2net[481]: Invalid port name/number: Invalid data to parameter on line 7 column 0
```
The service starts, so the restart on failure isn't doing anything.

The configuration file contains:
```
default:
   name: speed
   value: 115200n81

connection: &con00
  accepter: tcp,9100
  connector: serialdev,/dev/ttyUSB0,local
```
I think that at the time the service starts the USB port is not active yet.

If I then do this:
```
fabnav@docker-hass:~ $ sudo systemctl restart ser2net
fabnav@docker-hass:~ $ sudo systemctl status ser2net
● ser2net.service - Serial port to network proxy
     Loaded: loaded (/lib/systemd/system/ser2net.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2022-06-07 17:51:56 EDT; 2s ago
       Docs: man:ser2net(8)
   Main PID: 3062 (ser2net)
      Tasks: 1 (limit: 8985)
        CPU: 28ms
     CGroup: /system.slice/ser2net.service
             └─3062 /usr/sbin/ser2net -n -c /etc/ser2net.yaml -P /run/ser2net.pid

Jun 07 17:51:56 docker-hass systemd[1]: Starting Serial port to network proxy...
Jun 07 17:51:56 docker-hass systemd[1]: Started Serial port to network proxy.
```
Everything works.

**Reply by marcus-j-davies** [reply]
This might be an answer (but have not had any experience with it)
https://superuser.com/questions/1322777/in-systemd-service-file-how-do-i-say-after-usb-is-ready

**Comment by fabnavigator** [comment]
Thanks. I saw that, and tried some stuff, with no success.

**Comment by fabnavigator** [comment, marked as answer]
I added ```@reboot sleep 15 && systemctl restart ser2net``` to sudo crontab, and things work after a reboot.
